### PR TITLE
Bump .NET to 5.0.0-preview.4

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Bump to the latest version of .NET 5.0.0-preview.4 pre-release version.

## PR Context

Update to the latest SDK and packages from 5.0.0-preview.4

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: DotnetRuntimeMetadata.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (PoshChan) [inline DotnetRuntimeMetadata.json:1]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @{InputObject=System.Security.Principal.dll; SideIndicator==>}.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\Startup.Tests.ps1: line 119
119:         $diffs | Should -BeExactly $null
```

Comment (PoshChan) [inline DotnetRuntimeMetadata.json:1]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @{InputObject=System.Security.Principal.dll; SideIndicator==>}.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\Startup.Tests.ps1: line 119
119:         $diffs | Should -BeExactly $null
```

**Comment by adityapatwardhan** [comment]
@PoshChan please retry windows

**Comment by PoshChan** [comment]
@adityapatwardhan, successfully started retry of `PowerShell-CI-Windows`

File: DotnetRuntimeMetadata.json (hunk 1 above)
Comment (PoshChan) [inline DotnetRuntimeMetadata.json:1]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @{InputObject=System.Security.Principal.dll; SideIndicator==>}.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\Startup.Tests.ps1: line 119
119:         $diffs | Should -BeExactly $null
```

**Review by iSazonov — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@PoshChan please retry windows

**Comment by PoshChan** [comment]
@daxian-dbw, successfully started retry of `PowerShell-CI-Windows`

File: DotnetRuntimeMetadata.json (hunk 1 above)
Comment (PoshChan) [inline DotnetRuntimeMetadata.json:1]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @{InputObject=System.Security.Principal.dll; SideIndicator==>}.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\Startup.Tests.ps1: line 119
119:         $diffs | Should -BeExactly $null
```

**Comment by adityapatwardhan** [comment]
@daxian-dbw I am adding the new assembly to the allowed list. On a local build here is what is see:

```powershell
PS> [System.AppDomain]::CurrentDomain.GetAssemblies() | sls 'System.Security.Principal' -SimpleMatch

System.Security.Principal.Windows, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
System.Security.Principal, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
```

**Comment by huzefaMD** [comment]
t

**Comment by daxian-dbw** [comment]
@adityapatwardhan That looks fine to me. We will see how that goes after this PR is merged.

**Review by daxian-dbw — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@daxian-dbw Ready for merge?

**Comment by TravisEz13** [comment]
@PoshChan please rerun windows

**Comment by PoshChan** [comment]
@TravisEz13, successfully started rebuild of `PowerShell-CI-Windows`

**Comment by TravisEz13** [comment]
I think there is a conflict with another PR....

**Comment by daxian-dbw** [comment]
@adityapatwardhan is preview.4 released? I cannot find it at the official site https://dotnet.microsoft.com/download/dotnet/5.0

**Pull request merged**